### PR TITLE
Bh hackathon

### PR DESCRIPTION
**Pull request by benjaminha14**

### CODE REVIEW

File: src/ideas/decorations/SlotMachine/idea.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{
+  "id": "ac5855ff-d070-43a3-a29e-d18c66dbbac8",
```
Comment (alex-shortt) [inline src/ideas/decorations/SlotMachine/idea.json]: can you remove this field for the pr?

File: src/ideas/decorations/SlotMachine/idea.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    {
+      "name": "height",
+      "type": "number"
+    },
+    {
+      "name": "num_of_icons",
+      "type": "number"
+    },
+    {
+      "name": "num_plays_per_day",
```
Comment (alex-shortt) [inline src/ideas/decorations/SlotMachine/idea.json]: would suggest `playsPerDay`, there's not too much space on builder wand title currently and i have logic to convert camel case to lowercase + spaced out, i.e. "playsPerDay" --> "plays per day"

File: src/ideas/decorations/SlotMachine/idea.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    {
+      "name": "width",
+      "type": "number"
+    },
+    {
+      "name": "height",
+      "type": "number"
+    },
+    {
+      "name": "num_of_icons",
```
Comment (alex-shortt) [inline src/ideas/decorations/SlotMachine/idea.json]: same as below, i'd suggest `numOfIcons` or `numIcons`

File: src/ideas/decorations/SlotMachine/idea.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      "name": "picture_b",
+      "type": "image"
+    },
+    {
+      "name": "picture_c",
+      "type": "image"
+    },
+    {
+      "name": "picture_d",
+      "type": "image"
```
Comment (alex-shortt) [inline src/ideas/decorations/SlotMachine/idea.json]: same as above, prefer camel case to underscores

File: src/ideas/decorations/SlotMachine/index.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
+import { useState, Suspense, useEffect } from "react";
+import { GroupProps } from "@react-three/fiber";
+import { Text } from "@react-three/drei";
+import { Image } from "spacesvr";
+import Button from "./components/Button";
+
+export type SlotMachineProps = {
+  url?: string;
+  numReals?: number;
```
Comment (alex-shortt) [inline src/ideas/decorations/SlotMachine/index.tsx]: dont see this prop in the idea json

File: src/ideas/decorations/SlotMachine/idea.json [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+      "type": "rotation"
+    },
+    {
+      "name": "width",
+      "type": "number"
+    },
+    {
+      "name": "height",
+      "type": "number"
+    },
```
Comment (alex-shortt) [inline src/ideas/decorations/SlotMachine/idea.json]: this don't seem to be used?

File: src/ideas/decorations/SlotMachine/index.tsx [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+  const icons = [picture_a, picture_b, picture_c, picture_d];
+
+  const [visible, setVisible] = useState(false);
+  const [messageVisible, setMessageVisible] = useState(false);
+  const [real1, setReal1] = useState(0);
+  const [real2, setReal2] = useState(0);
+  const [real3, setReal3] = useState(0);
+
+  const slotGame = () => {
+    console.log("triggered");
```
Comment (alex-shortt) [inline src/ideas/decorations/SlotMachine/index.tsx:66]: remove console log unless necessary

File: src/ideas/decorations/SlotMachine/index.tsx [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+        <group name="realTwo">
+          <Suspense fallback={<Text> Loading</Text>}>
+            <Image transparent src={icons[real1]} position-x={1} />
+          </Suspense>
+
+          <Suspense fallback={<Text> Loading</Text>}>
+            <Image transparent src={icons[real2]} position-x={0} />
+          </Suspense>
+          <Suspense fallback={<Text> Loading</Text>}>
+            <Image transparent src={icons[real3]} position-x={-1} />
```
Comment (alex-shortt) [inline src/ideas/decorations/SlotMachine/index.tsx:119]: no longer need transparent prop on images, it's transparent by default

File: src/ideas/decorations/SlotMachine/index.tsx [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+      slotGame();
+      plays += 1;
+      setWithExpiry("playsToday", plays, 86400);
+    } else {
+      setMessageVisible(true);
+    }
+  };
+
+  return (
+    <group {...rest}>
```
Comment (alex-shortt) [inline src/ideas/decorations/SlotMachine/index.tsx]: add idea name as group name, `name="slot-machine"`

File: src/ideas/decorations/SlotMachine/index.tsx [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+          position-z={1}
+          onClick={() => window.open(url, "_blank")}
+        />
+      ) : undefined}
+      {messageVisible ? (
+        <Text position-x={0} position-y={-0.4} position-z={1} color="red">
+          You are out of plays. Try again tomorrow.
+        </Text>
+      ) : undefined}
+      ;
```
Comment (alex-shortt) [inline src/ideas/decorations/SlotMachine/index.tsx]: dangling semicolon

**Review by alex-shortt — CHANGES_REQUESTED** [review]
please update pr name to the idea name

and the description to include some useful info, this is kinda our docs for now. example: https://github.com/musehq/muse-ideas/pull/10#issue-1362464654

File: src/ideas/decorations/SlotMachine/idea.json (hunk 6 above)
Comment (benjaminha14) [reply src/ideas/decorations/SlotMachine/idea.json]: not sure how to do it but i would like to allow the slot machine to be rescaled

**Pull request closed without merging**